src/OpenTK/MouseCursor.cs

```diff
@@ -28,28 +28,55 @@
 #endregion
 
 using System;
+using System.Collections;
+using System.Collections.Generic;
 
 namespace OpenTK
 {
     /// <summary>
+    /// Stores information about frame timing
+    /// </summary>
+    public struct MouseCursorFrameInformation
+    {
+        /// <summary>
+        /// The frames time in seconds
+        /// </summary>
+        public float duration;
+
+        /// <summary>
+        /// The associated frame
+        /// </summary>
+        public MouseCursorFrame frame;
+    }
+
+    /// <summary>
     /// Represents a predefined or custom mouse cursor.
+    /// new MouseCursor() represents an empty animation. Use MouseCursor.Default instead.
     /// </summary>
-    public sealed class MouseCursor : WindowIcon
+    public sealed class MouseCursor : IEnumerable<MouseCursorFrameInformation>
     {
         static readonly MouseCursor default_cursor = new MouseCursor();
-        static readonly MouseCursor empty_cursor = new MouseCursor(
-            0, 0, 16, 16, new byte[16 * 16 * 4]);
+        static readonly MouseCursor empty_cursor = new MouseCursor(0, 0, 16, 16, new byte[16 * 16 * 4]);
```

Comment:
No Idea, just encapsulated the original variant from
https://github.com/opentk/opentk/blob/d50c2fcea0696d3cc82aeb339e15c049405129ba/src/OpenTK/MouseCursor.cs